docs/knowledge_base/tips/using_tables/index.md

```diff
@@ -0,0 +1,327 @@
+    col = payload["col"] # column name
+    value = payload["value"] # new value cast to the column type
+    user_value = payload["user_value"] # new value as entered by the user
+ 
+    # state.food_df.loc[index, col] = value #  Don't do this!
+    old_value = state.food_df.loc[index, col]
+    new_food_df = state.food_df.copy()
+    new_food_df.loc[index, col] = value
+    state.food_df = new_food_df
+    notify(state, "I", f"Edited value from '{old_value}' to '{value}'. (index '{index}', column '{col}')")
+ 
+main_md = Markdown("<|{food_df}|table|on_edit=food_df_on_edit|>")
+```
+
+The table documentation provides more information on the function signature 
+which is slightly different for each data modification property. 
+The code example above is self-explanatory though.
+
+Notice that we didn’t modify the `food_df` DataFrame with `state.food_df.loc[index, col] = value`. 
+This is because Taipy state variables should be updated by assignment to the variable itself 
+hence why we instead create a copy of the DataFrame, modify the relevant cell, then assign it back to `state.food_df`.
+
+## Adding (on_add)
+
+Adding and deleting are quite similar to editing. When you specify the *on_add* property, 
+a *button* with a plus icon is included, and when clicked, it triggers the defined *on_add* callback function.
+
+![Adding](tables-on_add.gif){width=100%}
+
+We can implement the functionality above as follows:
+
+```py
+def food_df_on_add(state, var_name, action, payload):
+    empty_row = pd.DataFrame([[None for _ in state.food_df.columns]], columns=state.food_df.columns)
+    state.food_df = pd.concat([empty_row, state.food_df], axis=0, ignore_index=True)
+ 
+    notify(state, "S", f"Added a new row.")
+ 
+main_md = Markdown("<|{food_df}|table|on_add=food_df_on_add|>")
+```
+
+This code simply adds a new empty row to the top of the table (DataFrame). 
+You can customize the callback function accordingly if your use case requires 
+some columns to have a default or set value.
+
+## Deleting (on_delete)
+
+Finally, the deletion process works as follows:
+
+![Deleting](tables-on_delete.gif){width=100%}
+
+We can implement basic functionality with the following code:
+
+```py
+def food_df_on_delete(state, var_name, action, payload):
+    index = payload["index"] # row index
+ 
+    state.food_df = state.food_df.drop(index=index)
+    notify(state, "E", f"Deleted row at index '{index}'")
+ 
+main_md = Markdown("<|{food_df}|table|on_delete=food_df_on_delete|>")
+```
+
+## The Complete App
+
+Once more, please note that this tip article doesn't cover all the features of tables 
+in Taipy comprehensively. If you're seeking a feature that we didn't address here, 
+be sure to refer to the [documentation](../../../manuals/gui/viselements/table/) for more information!
+
+Lastly, here's the code that combines all the features we discussed in this article, 
+used to create the app shown at the beginning of the article:
+
+```py
```

Comment:
```suggestion
```python linenums="1"
```

Reply:
Drop linenums="1" (you don't use those line numbers)